src/bundle/Resources/public/js/scripts/fieldType/ezimageasset.js

```diff
@@ -0,0 +1,240 @@
+            });
+
+            this.updateData(response.ContentInfo.Content._id, imageField.fieldValue);
+            this.showPreview();
+        }
+
+        /**
+         * Open UDW to select an existing Image Asset
+         *
+         * @method openUDW
+         * @param {Event} event
+         */
+        openUDW(event) {
+            const udwContainer = doc.getElementById('react-udw');
+            const config = JSON.parse(event.currentTarget.dataset.udwConfig);
+            const title = Translator.trans(/*@Desc("Select Image Asset")*/ 'ezimageasset.title', {}, 'universal_discovery_widget');
+
+            const closeUDW = () => ReactDOM.unmountComponentAtNode(udwContainer);
+            const onCancel = closeUDW;
+            const onConfirm = (items) => {
+                closeUDW();
+                this.loadPreview(items[0]);
+            };
+            const canSelectContent = ({item, itemsCount}, callback) => {
+                const itemContentType = item.ContentInfo.Content.ContentTypeInfo.identifier;
+                const isAllowedContentType = itemContentType === imageAssetMapping['content_type_identifier'];
+
+                callback(isAllowedContentType);
+            };
+
+            ReactDOM.render(React.createElement(eZ.modules.UniversalDiscovery, Object.assign(config, {
+                onConfirm,
+                onCancel,
+                canSelectContent,
+                title,
+                multiple: false,
+                startingLocationId: parseInt(imageAssetMapping['parent_location_id'], 10),
+                restInfo: {token, siteaccess},
+            })), udwContainer);
+        }
+
+        /**
+         * Checks if file size is an allowed limit
+         *
+         * @method handleInputChange
+         * @param {Event} event
+         */
+        handleInputChange(event) {
+            const file = event.currentTarget.files[0];
+            const languageCode = event.currentTarget.dataset.languageCode;
+            const isFileSizeLimited = this.maxFileSize > 0;
+            const maxFileSizeExceeded = isFileSizeLimited && file.size > this.maxFileSize;
+
+            if (maxFileSizeExceeded) {
+                this.resetInputField();
+                return;
+            }
+
+            this.fieldContainer.querySelector('.ez-field-edit__option--remove-media').checked = false;
+
+            this.createImage(file, languageCode);
+        }
+
+        /**
+         * Resets input field state
+         *
+         * @method resetInputField
+         */
+        resetInputField() {
+            super.resetInputField();
+
+            this.inputDestinationContentId.value = '';
+        }
+
+        /**
+         * Initializes the preview
+         *
+         * @method init
+         */
+        init() {
+            super.init();
+
+            this.btnSelect = this.fieldContainer.querySelector('.ez-data-source__btn-select');
+            this.btnSelect.addEventListener('click', (event) => {
```

Comment:
I do not see a point of this middle function which will only `preventDefault`, you can do it in `openUDW`